### PR TITLE
fix: FIR-36807 command timeout parameter is not working in net

### PR DESCRIPTION
**Pull request by stepansergeevitch**
Fixed async execution functions to account for the command timeout value, that's set for a command

### CODE REVIEW

File: FireboltDotNetSdk.Tests/Integration/ExecuteTest.cs [hunk 1]
```diff
@@ -121,7 +123,44 @@ public void WithTimeout(int? timeout)
...
+
+        [Timeout(20)]
+        public void TimeoutThrowsError()
+        {
+            using var conn = new FireboltConnection(SYSTEM_CONNECTION_STRING);
+            conn.Open();
+            DbCommand command = conn.CreateCommand();
+            command.CommandText = LONG_QUERY;
+            command.CommandTimeout = 10;
         }
```
Comment (ptiurin) [inline FireboltDotNetSdk.Tests/Integration/ExecuteTest.cs:164]: Is this used anywhere?

File: FireboltNETSDK/Client/FireboltCommand.cs [hunk 2]
```diff
@@ -178,15 +178,19 @@ private DbDataReader CreateDbDataReader(QueryResult? queryResult)
             {
                 commandText = ValidateSetCommand(commandText.Remove(0, 4).Trim());
                 SetParamList.Add(commandText);
-                try
+                if (Connection != null)
```
Comment (ptiurin) [inline FireboltNETSDK/Client/FireboltCommand.cs:181]: Do we need this check? We already verify Connection is not null in https://github.com/firebolt-db/firebolt-net-sdk/pull/106/files#diff-9f5e4d8a0218e3f437d1707871b746709b804521e38023b2328a94ffb378452aR168

**Review by ptiurin — APPROVED** [review]
Couple of comments, otherwise looks good

File: FireboltNETSDK/Client/FireboltCommand.cs (hunk 2 above)
Comment (stepansergeevitch) [reply FireboltNETSDK/Client/FireboltCommand.cs:181]: Good point, I did this for type checker

File: FireboltDotNetSdk.Tests/Integration/ExecuteTest.cs (hunk 1 above)
Comment (stepansergeevitch) [reply FireboltDotNetSdk.Tests/Integration/ExecuteTest.cs:164]: No, this is a leftover

**Pull request merged**